pkg/cmd/server/origin/ensure.go

```diff
@@ -45,8 +42,16 @@ func (c *MasterConfig) ensureOpenShiftInfraNamespace() {
 
 	// Ensure namespace exists
 	namespace, err := c.KubeClient().Namespaces().Create(&kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: ns}})
-	if err != nil && !kapierror.IsAlreadyExists(err) {
+	if kapierror.IsAlreadyExists(err) {
+		// Get the persisted namespace
+		namespace, err = c.KubeClient().Namespaces().Get(ns)
```

Comment:
in the case where we got a AlreadyExists error, we were incorrectly using the returned namespace, which had no name set, instead of fetching the persisted namespace